crates/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -44,50 +44,63 @@ pub trait Environment {
     async fn build(
         &mut self,
         nix: &NixCommandLine,
-        system: impl AsRef<str> + Send,
+        system: System,
     ) -> Result<(), EnvironmentError2>;
 
     /// Install packages to the environment atomically
     async fn install(
         &mut self,
-        packages: impl IntoIterator<Item = FloxPackage> + Send,
+        packages: Vec<FloxPackage>,
         nix: &NixCommandLine,
-        system: impl AsRef<str> + Send,
+        system: System,
     ) -> Result<bool, EnvironmentError2>;
 
     /// Uninstall packages from the environment atomically
     async fn uninstall(
         &mut self,
-        packages: impl IntoIterator<Item = FloxPackage> + Send,
+        packages: Vec<FloxPackage>,
         nix: &NixCommandLine,
-        system: impl AsRef<str> + Send,
+        system: System,
     ) -> Result<bool, EnvironmentError2>;
 
     /// Atomically edit this environment, ensuring that it still builds
     async fn edit(
         &mut self,
         nix: &NixCommandLine,
-        system: impl AsRef<str> + Send,
-        contents: impl AsRef<str> + Send,
+        system: System,
+        contents: String,
     ) -> Result<(), EnvironmentError2>;
 
     async fn catalog(
         &self,
         nix: &NixCommandLine,
-        system: impl AsRef<str> + Send,
+        system: System,
     ) -> Result<EnvCatalog, EnvironmentError2>;
 
+    /// Extract the current content of the manifest
+    fn manifest_content(&self) -> Result<String, EnvironmentError2>;
+
     /// Return the [EnvironmentRef] for the environment for identification
     fn environment_ref(&self) -> &EnvironmentRef;
 
+    /// Return a flake attribute installable for this environment
+    fn flake_attribute(&self, system: System) -> FlakeAttribute;
+
     /// Returns the environment owner
     fn owner(&self) -> Option<EnvironmentOwner>;
 
     /// Returns the environment name
     fn name(&self) -> EnvironmentName;
 
     /// Delete the Environment
-    fn delete(self) -> Result<(), EnvironmentError2>;
+    fn delete(self) -> Result<(), EnvironmentError2>
+    where
+        Self: Sized;
+
+    /// Remove gc-roots
+    fn delete_symlinks(&self) -> Result<bool, EnvironmentError2> {
```

Comment:
Would we ever use this for `PathEnvironment`?

Reply:
Maybe not, it was one of the methods that wasn't captured by the trait.
We might eventually make it a separate trait implemented for managed and remote envs only (do remote envs have this?).

Here, its mainly for it to have a place in the meantime

Reply:
The semantics of this method are also weirdly specific and not quite captured in either the name or comment. I'd like to revisit this anyway considering this impl assumes we run `nix store gc` which is still not the most precise 🤔

Reply:
```suggestion
    // TODO consider renaming or removing - we might not support this for PathEnvironment
    fn delete_symlinks(&self) -> Result<bool, EnvironmentError2> {
```